packages/opentelemetry-core/src/trace/sampler/ProbabilitySampler.ts

```diff
@@ -23,6 +23,12 @@ export class ProbabilitySampler implements Sampler {
   }
 
   shouldSample(parentContext?: SpanContext) {
+    // Respect the parent sampling decision if there is one
+    if (parentContext && parentContext.traceFlags !== undefined) {
```

Comment:
this might be tricky if any1 would ever defined / override an `undefined` , if instead you choose `typeof parentContext.traceFlags !== 'undefined'` it will never be broken.

Reply:
Did not know that was possible. Browsers are weird

Reply:
at least in node this seems to be not possible. What if someone overrides `typeof`?
I would prefer `!= null` in general as it includes `null` and `undefined`.